pkg/proxy/configs/configs.go

```diff
@@ -58,6 +59,13 @@ func CreateProxyConfigs(args ArgsProxyConfigs) (*ArgsOutputConfig, error) {
 			Address:  httpPrefix + nodeAPIInterface,
 			IsSynced: true,
 		})
+		if args.IsSovereign {
```

Comment:
I know this is cumbersome and you need extra code + runTypeComps, but adding a `bool` flag that changes behavior within a component is considered bad design code.
It can stay as it is for now if it's too much for this PR, but please add a task and TODO here to refactor it.

Reply:
Applies to `NewConfigsFetcher` as well

Reply:
Create task MX-16540 for RunType components